src/platform/K32W/Logging.cpp

```diff
@@ -0,0 +1,248 @@
+            memcpy(buf, "[Error]", 7);
+            break;
+        case kLogCategory_Progress:
+        case kLogCategory_Retain:
+        default:
+            memcpy(buf, "[Progress]", 10);
+            break;
+        case kLogCategory_Detail:
+            memcpy(buf, "[Debug]", 7);
+            break;
+        }
+    }
+
+    if (otLogLevel != OT_LOG_LEVEL_NONE)
+    {
+        switch (otLogLevel)
+        {
+        case OT_LOG_LEVEL_CRIT:
+            memcpy(buf, "[Error]", 7);
+            break;
+        case OT_LOG_LEVEL_WARN:
+            memcpy(buf, "[Warn]", 6);
+            break;
+        case OT_LOG_LEVEL_NOTE:
+        case OT_LOG_LEVEL_INFO:
+        default:
+            memcpy(buf, "[Info]", 6);
+            break;
+        case OT_LOG_LEVEL_DEBG:
+            memcpy(buf, "[Debug]", 7);
+            break;
+        }
+    }
+}
+
+void FillPrefix(char * buf, uint8_t bufLen, uint8_t chipCategory, uint8_t otLogLevel, uint8_t module)
+{
+    size_t prefixLen;
+
+    /* add the error string */
+    assert(bufLen > nlChipLoggingChipPrefixLen);
+    ::GetMessageString(buf, chipCategory, otLogLevel);
+
+    /* add the module name string */
+    prefixLen = strlen(buf);
+    assert(bufLen > (prefixLen + nlChipLoggingModuleNameLen + 3));
+    buf[prefixLen++] = '[';
+    GetModuleName(buf + prefixLen, module);
+    prefixLen        = strlen(buf);
+    buf[prefixLen++] = ']';
+    buf[prefixLen++] = ' ';
+}
+
+} // unnamed namespace
+
+namespace chip {
+namespace DeviceLayer {
+
+/**
+ * Called whenever a log message is emitted by CHIP or LwIP.
+ *
+ * This function is intended be overridden by the application to, e.g.,
+ * schedule output of queued log entries.
+ */
+void __attribute__((weak)) OnLogOutput(void) {}
+
+} // namespace DeviceLayer
+} // namespace chip
+
+namespace chip {
+namespace Logging {
+
+/**
+ * CHIP log output function.
+ */
+void LogV(uint8_t module, uint8_t category, const char * msg, va_list v)
+{
+    (void) module;
+    (void) category;
+
+#if K32W_LOG_ENABLED
+
+    if (IsCategoryEnabled(category))
+    {
+        {
+            char formattedMsg[CHIP_DEVICE_CONFIG_LOG_MESSAGE_MAX_SIZE - 1] = { 0 };
+            size_t prefixLen, writtenLen;
+
+            /* Prefix is composed of [Debug String][MOdule Name String] */
+            FillPrefix(formattedMsg, CHIP_DEVICE_CONFIG_LOG_MESSAGE_MAX_SIZE - 1, category, OT_LOG_LEVEL_NONE, module);
+            prefixLen = strlen(formattedMsg);
+
+            // Append the log message.
+            writtenLen = vsnprintf(formattedMsg + prefixLen, sizeof(formattedMsg) - prefixLen - EOL_CHARS_LEN, msg, v);
+            assert(writtenLen > 0);
+            memcpy(formattedMsg + prefixLen + writtenLen, EOL_CHARS, EOL_CHARS_LEN);
+
+            K32WWriteBlocking((const uint8_t *) formattedMsg, strlen(formattedMsg));
+        }
+    }
+
+    // Let the application know that a log message has been emitted.
+    DeviceLayer::OnLogOutput();
+
+#endif // NRF_LOG_ENABLED
+}
```

Comment:
Should probably remove the NRF names that are not aligned with the K32W names

Reply:
Will remove the comment.